packages/browser/src/core/http-cookies/index.ts

```diff
@@ -35,22 +36,37 @@ export class HTTPCookieService {
   private flushIntervalId?: NodeJS.Timer
 
   private constructor(options: HTTPCookieServiceOptions) {
-    this.renewUrl = options.renewUrl
-    this.clearUrl = options.clearUrl
+    const urls = HTTPCookieService.urlHelper(options)
+    this.renewUrl = urls.renewUrl
+    this.clearUrl = urls.clearUrl
+
     this.backoff = options.backoff ?? 300
     this.retries = options.retries ?? 3
     this.flushInterval = options.flushInterval ?? 1000
     this.queue = []
   }
 
+  static urlHelper(options: HTTPCookieServiceOptions): {
+    origin: string
+    renewUrl: string
+    clearUrl: string
+  } {
+    const origin = options.origin ?? new URL(window.location.href).origin
```

Comment:
can we use `window.location.origin` here or is constructing a `new URL` necessary?

Reply:
We probably can 👍